src/librustc_mir/monomorphize/collector.rs

```diff
@@ -1081,15 +1081,27 @@ fn create_mono_items_for_default_impls<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
                         continue;
                     }
 
-                    let instance = ty::Instance::resolve(tcx,
-                                                         ty::ParamEnv::empty(traits::Reveal::All),
-                                                         method.def_id,
-                                                         callee_substs).unwrap();
-
-                    let mono_item = create_fn_mono_item(instance);
-                    if mono_item.is_instantiable(tcx)
-                        && should_monomorphize_locally(tcx, &instance) {
-                        output.push(mono_item);
+                    let fn_ty = tcx.erase_regions(&tcx.type_of(method.def_id));
```

Comment:
Would it be better to erase regions after applying the trait substs?

Reply:
Maybe if it makes the code simpler.